### ISSUE TITLE
Implement `hashValue` in IGSDB

### ISSUE THREAD

**Issue by christoph-maurer** [building-envelope-data/api#330]
We have two software companies who plan to use https://igsdb-v2.herokuapp.com/graphql/ in the next months to access the optical data of glass panes and use it for building performance simulation and to determine the properties of complex fenestration systems. They store the data in their applications and need to check in regular intervals if there are any changes.

@danielmcquillen How much effort would it be for you to implement the field [hashValue](https://github.com/building-envelope-data/api/blob/c9c2c3d9098d3f9a2a0698cc7c76bf2c30b41d91/apis/database.graphql#L1806-L1862)? It could make the process of updating the data sets much easier for the companies, because they only need to download data sets with another `hashValue` than they already have.

**Comment by danielmcquillen** [comment]
@christoph-maurer I will make sure to raise this issue in a meeting tomorrow with @RDmitchell

**Comment by christoph-maurer** [comment]
Thank you @danielmcquillen ! I am fully aware of the limited resource and the need to prioritize. As always, the issues raised by me are open for discussion.

**Comment by RDmitchell** [comment]
@christoph-maurer -- we don't have the bandwidth or the funding to work on further development of our graphQL API for you. 

However, you are welcome to work on it to implement what you need.

We can add you to our repository as a Contributor, you can develop the needed code, and submit a PR for our review. Then hopefully we can implement it and you will be able to add the functionality you are looking for. 

Would this work for you?

**Comment by christoph-maurer** [comment]
I need to discuss it with my colleague next time when I see him in September.

**Comment by christoph-maurer** [comment]
@RDmitchell Our funding is limited, too, and we have no experience with the source code of the IGSDB yet. But yes, please add @simon-wacker and me as Contributors. We will have a look at the source code and the issues to check what may be feasible.

**Comment by danielmcquillen** [comment]
Hi @christoph-maurer both you and Simon are already contributors on the igsdb-webapp github repo. I've updated both users to have write access, so you should be able to create PRs. I'll try to add some extra docs and perhaps some wiki pages to explain anything that might not be immediately obvious (e.g. https://github.com/LBNL-ETA/igsdb-webapp/wiki/Project-Branches)

**Comment by christoph-maurer** [comment]
Thank you, @danielmcquillen !

**Comment by danielmcquillen** [comment]
@christoph-maurer @simon-wacker Did you ever implement the hash value you need within the GraphQL api?

**Comment by simon-wacker** [comment]
@danielmcquillen  Yes, this is what lives partly in https://github.com/LBNL-ETA/igsdb-webapp/blob/main/igsdb_webapp/digest/factories.py now. And both the staging and production version of the IGSDB return the correct hash values when I ask for them

```
query {
  allOpticalData {
    edges {
      node {
        resources {
          hashValue
          locator
        }
      }
    }
  }
}
```

Great!
